### DISCUSSION TITLE
DDT - gestione magazzino via API

### DISCUSSION THREAD

**Discussion by lorenzoSIT** [fattureincloud/api#98] [Q&A] [answered]
Ciao a tutti,

vorrei utilizzare le API per la gestione del magazzino.
Ho guardato la documentazioni ma non ho visto la possibilità di creare, ottenere movimenti secchi di magazzino. 
L'unico modo che ho visto per movimentare (solo in scarico) il magazzino è tramite DDT giusto? 

Dal sito ho visto che si può caricare il magazzino registrando degli "Acquisti"; però dalle API non ho trovato il modo, quindi la domanda è: è possibile registrare acquisti che carichino il magazzino tramite API? 

Grazie 
Ciao

**Comment by emajo** [comment, marked as answer]
Ciao @lorenzoSIT attualmente non abbiamo documentato i metodi per la movimentazione del magazzino, ma come dici tu è possibile farlo registrando un acquisto, mettendo all'interno degli items dei `items_list` il campo `stock = true`.

Es.
```
        "items_list": [
            {
                "product_id": 1584648,
                "code": "code",
                "description": "",
                "discount": 0,
                "amount_gross": 0,
                "name": "prod name",
                "net_price": 10,
                "qty": 10,
                "vat": {
                    "id": 0
                },
                "stock": true,
                "order": 1
            }
        ]
```

**Reply by lorenzoSIT** [reply]
> Ciao @lorenzoSIT attualmente non abbiamo documentato i metodi per la movimentazione del magazzino, ma come dici tu è possibile farlo registrando un acquisto, mettendo all'interno degli items dei `items_list` il campo `stock = true`.
> 
> Es.
> 
> ```
>         "items_list": [
>             {
>                 "product_id": 1584648,
>                 "code": "code",
>                 "description": "",
>                 "discount": 0,
>                 "amount_gross": 0,
>                 "name": "prod name",
>                 "net_price": 10,
>                 "qty": 10,
>                 "vat": {
>                     "id": 0
>                 },
>                 "stock": true,
>                 "order": 1
>             }
>         ]
> ```

Ciao @emajo e grazie per la tua risposta; 
mi confermi che per registrare un Acquisto è necessario usare la struttura dell'oggetto Received Documents ? 
Grazie

**Comment by Picaland** [comment]
Buongiorno a tutti,
mi aggangio a questa discussione evitando di aprirene un'altra.

Vorrei fare la stessa cosa ma durante la creazione della fattura. 
https://developers.fattureincloud.it/api-reference/#post-/c/-company_id-/issued_documents

Vorrei sapere se tramite items_list come nell'esempio:

```
"items_list": [
      {
        "product_id": 333,
        "code": "SG3",
        "name": "Soggiorno",
        "measure": "",
        "net_price": 68.18182,
        "category": "",
        "id": 277875565,
        "gross_price": 75,
        "apply_withholding_taxes": true,
        "discount": 0,
        "discount_highlight": false,
        "in_dn": false,
        "qty": 1,
        "vat": {
          "id": 3,
          "value": 10,
          "description": ""
        },
        "stock": true,
        "description": "",
        "not_taxable": false
      }
    ],
```
passando stock = true e l'id del prodotto abbiamo una movimentazione del magazzino ?

Grazie!

**Reply by emajo** [reply]
Ciao @Picaland, esatto facendo così verrà movimentato il magazzino in scarico, mentre creando un acquisto viene movimentato in carico.

**Reply by Picaland** [reply]
@emajo mi è venuto un dubbio, serve per caso avere i permessi abilitati ?
stock:a - Write permission to the Stock movements

**Reply by emajo** [reply]
si, servono quei permessi

**Reply by Picaland** [reply]
Grazie @emajo
